packages/edit-site/src/components/global-styles/font-size.js

```diff
@@ -0,0 +1,239 @@
+	__experimentalView as View,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	privateApis as componentsPrivateApis,
+	Button,
+	FlexItem,
+	ToggleControl,
+} from '@wordpress/components';
+import { useRef } from '@wordpress/element';
+import { moreVertical } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+const {
+	SizeControl,
+	DropdownMenuV2: DropdownMenu,
+	DropdownMenuItemV2: DropdownMenuItem,
+	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
+} = unlock( componentsPrivateApis );
+const { useGlobalSetting, useGlobalStyle } = unlock( blockEditorPrivateApis );
+import ScreenHeader from './header';
+
+function FontSizePreview( { fontSize } ) {
+	const [ font ] = useGlobalStyle( 'typography' );
+	return (
+		<div
+			className="edit-site-typography-preview"
+			style={ {
+				fontSize: fontSize.size,
+				fontFamily: font?.fontFamily ?? 'serif',
+			} }
+		>
+			{ __( 'Aa' ) }
+		</div>
+	);
+}
+
+function FontSize() {
+	const {
+		params: { slug },
+		goBack,
+	} = useNavigator();
+	const [ fontSizes, setFontSizes ] = useGlobalSetting(
+		'typography.fontSizes'
+	);
+	// Get the font sizes from the theme or use the default ones.
+	const sizes = fontSizes.theme ?? fontSizes.default;
+
+	// Get the font size by slug.
+	const fontSize = sizes.find( ( size ) => size.slug === slug );
+
+	const hasUnits = typeof fontSize.size === 'string';
+
+	// Whether fluid is true or an object, set it to true, otherwise false.
+	const isFluid = !! fontSize.fluid ?? false;
+
+	// Whether custom fluid values are used.
+	const isCustomFluid = typeof fontSize.fluid === 'object';
+
+	// Initialize the ref with the prop value on the first render
+	const initialFontSizeRef = useRef( fontSize.size );
+
+	// Memoized initial value of fontSize
+	const initialFontSize = initialFontSizeRef.current;
+
+	const handleNameChange = ( value ) => {
+		updateFontSize( 'name', value );
+	};
+
+	const handleFontSizeChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? initialFontSize;
+		updateFontSize( 'size', newValue );
+	};
+
+	const handleFluidChange = ( value ) => {
+		updateFontSize( 'fluid', value );
+	};
+
+	const handleCustomFluidValues = ( value ) => {
+		if ( value ) {
+			// If custom values are use init the values with the current ones.
+			updateFontSize( 'fluid', {
+				min: fontSize.size,
+				max: fontSize.size,
+			} );
+		} else {
+			// If custom fluid values are disabled, set fluid to true.
+			updateFontSize( 'fluid', true );
+		}
+	};
+
+	const handleMinChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? fontSize.size;
+		updateFontSize( 'fluid', { ...fontSize.fluid, min: newValue } );
+	};
+
+	const handleMaxChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? fontSize.size;
```

Comment:
I don't think `fontSize.size` is the initial value for `fluid`. As with this comment, is this process unnecessary?

Reply:
I removed all the reset related logic here: [`cc582db` (#62328)](https://github.com/WordPress/gutenberg/pull/62328/commits/cc582dbed6b82f512c542ec391fa1ae22de337b3)